### PR TITLE
[STRAT 149] CSV file for lead times and prices

### PR DESCRIPTION
**Pull request by rodrigotiscareno**

### CODE REVIEW

File: displays/parseCSV.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import pandas as pd
+
+def parseCSV(csv1, csv2, partNumber):
+
+    df1 = pd.read_csv(csv1)
+    df2 = pd.read_csv(csv2)
+
+    try:
+        value = df1.loc[df1['Digi-Key Part Number'] == partNumber, "Mfg Std Lead Time"].values[0]
```
Comment (e-wai) [inline displays/parseCSV.py:15]: + Do we know for sure that there will only be one listing of the part number (ie. unique entry, and the filtering by `[0]` doesn't hide anything)?
+ Is the formatting for the part number consistent across both? (@mdshiozaki - do we have to parse around dashes or anything?)

File: displays/parseCSV.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Requirements: 
+# - Files must be in CSV formats. 
+# - Two CSV's must be entered into the function, with the first csv containing a "Digi-Key Part Number" column value and a "Mfg Std Lead Time" value. The second CSV should contain a "Mouser Part Number" column value and a "Lead Time in Weeks" value. 
+# - The third parameter is either the Digi-Key Part Number or the Mouser Part Number. The part number is to be entered as a string, as with the other CSV names.
```
Comment (e-wai) [inline displays/parseCSV.py:4]: Nice to see, but probably not useful anymore?

File: displays/parseCSV.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    except:
+        print(f"Did not find part number in {csv1}.")
+
+    try:
+        value2 = df2.loc[df2['Mouser Part Number'] == partNumber, "Lead Time in Weeks"].values[0]
+        print(f"The lead time in weeks is: {value2}.")
+    except:
+        print(f"Did not find part number in {csv2}.")
+
+# Testing - Assuming names for the CSV files and selecting random part number from BomSample_.csv
```
Comment (e-wai) [inline displays/parseCSV.py:26]: nitpicky, but this comment doesn't tell us much and can probably be deleted?

File: displays/parseCSV.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        print(f"Did not find part number in {csv1}.")
+
+    try:
+        value2 = df2.loc[df2['Mouser Part Number'] == partNumber, "Lead Time in Weeks"].values[0]
+        print(f"The lead time in weeks is: {value2}.")
+    except:
+        print(f"Did not find part number in {csv2}.")
+
+# Testing - Assuming names for the CSV files and selecting random part number from BomSample_.csv 
+parseCSV('BomSample_.csv','MidnightSun-Jan28-QuoteRequest.csv','490-4789-1-ND')
```
Comment (e-wai) [inline displays/parseCSV.py:27]: Just to be careful, can you throw this under a `if __name__=='__main':`

**Review by e-wai — CHANGES_REQUESTED** [review]

**Review by e-wai — CHANGES_REQUESTED** [review]
Hey, so an updated list of constraints from Kristen:

+ Both the Mouse and Digikey lists are exhaustive of all the parts we want -> both lists are of equal lengths, and each have a row per item we're interested in. The number is listed under `Manufacturer Part Number` for Digikey and `Mfr. #` for Mouser. _However_, we need to remove dashes in the dashes in order to compare against both.

+ We're not just entering a single partNumber, but rather creating two updated lists ->  a list of all the parts that Digikey wins (according to the last bullet point's logic) and a list of al lthe parts that Mouser wins. Save these as two separate CSV files, please, and include the price and lead time details!

+ For each, the logic we want is that the lesser price wins UNLESS the lead time > 15 weeks (and let's keep 15 weeks as a parameter in case we want to change in the future)

File: displays/parseCSV.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            if Bom_prices[i]>MS_prices[i]:
+                MS_Quote_Winners.append(i)
+            else:
+                Bom_Winners.append(i)
+        elif Bom_lead[i]>15 & MS_lead[i]<=15:
+            MS_Quote_Winners.append(i)
+        elif Bom_lead[i]<=15 & MS_lead[i]>15:
+            Bom_Winners.append(i)
+        else:
+            Bom_Winners.append(i)
```
Comment (e-wai) [inline displays/parseCSV.py:53]: When is this case reached?

**Review by e-wai — COMMENTED** [review]
Hey sorry for the delay - has anyone reached out to talk about adding a `priority` field on the inputs? (If not, maybe we can talk on Slack?)

File: utility/parseCSV.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
+# Requirements: 
+# 1. Entering data spreadsheets to the function in order as referenced.
+# 2. Updating priority field on both spreadsheets, ensuring value is equal on both. 
+
+import pandas as pd
+
+# parseCSV will accept the BOM.csv and MS_Quote Excel File and return two seperate Excel/CSV files with winning parts. 
+def parseCSV(Bom, MS_Quote):
```
Comment (e-wai) [inline utility/parseCSV.py:8]: See overall comment - rename this too please!

Comment (e-wai) [reply utility/parseCSV.py:8]: + also can we rename the variables to bom_file and MS_quote_file? more descriptive

File: utility/parseCSV.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+# 1. Entering data spreadsheets to the function in order as referenced.
+# 2. Updating priority field on both spreadsheets, ensuring value is equal on both. 
+
+import pandas as pd
+
+# parseCSV will accept the BOM.csv and MS_Quote Excel File and return two seperate Excel/CSV files with winning parts. 
+def parseCSV(Bom, MS_Quote): 
+
+    Bom_DF = pd.read_csv(Bom)
+    MS_Quote_DF = pd.read_excel(MS_Quote)
```
Comment (e-wai) [inline utility/parseCSV.py:11]: Thinking that we should check with Kristen whether we will always get first script in csv, and second in excel. Feels like something we might trip on in future uses

File: utility/parseCSV.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    Bom_Win = result.loc[Bom_Winners]
+    MS_Win = result.loc[MS_Quote_Winners]
+    Bom_Win=Bom_Win.drop(Bom_Win[['Order Unit Price','Lead Time in Weeks']],1)
+    MS_Win = MS_Win.drop(MS_Win[['Unit Price','Mfg Std Lead Time']],1)
+
+    Bom_Win.to_excel('Bom_Parts.xlsx')
+    MS_Win.to_excel('MS_Quote_Parts.xlsx')
+
+if __name__ == "__main__":
+    parseCSV('Bom.csv', 'MS Quote Request.xls')
```
Comment (e-wai) [inline utility/parseCSV.py:83]: Can we add another line below here to verify something happened? As simple as

```python
    print("Wrote selected purchases to Bom_Parts.xlsx and MS_Quote_Parts.xlsx")
```

**Review by e-wai — CHANGES_REQUESTED** [review]
Looks great 🎉 

Couple of last details:
- When I run it, there are a couple dependencies that I'm missing. Can we add `xlrd`, `openpyxl` to the `requirements.txt`? I can also check with Kristen whether all CSVs are used, because that's preferable imo
- Can we create a subdirectory within `utility` to change the file paths -> `utility/hardware_bom`
- Can we rename `parseCSV.py` to something more specific + descriptive, maybe `select_hardware_bom_purchases.py` or `hardware_bom_purchases.py` or `parse_hardware_bom.py`?

**Comment by rodrigotiscareno** [comment]
Please refer to new PR [STRAT 149] CSV file for lead times and prices.

**Pull request closed without merging**